File: leetcode/1.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val indicies =
            for {
                a <- 0 until nums.length - 1
                b <- a + 1 until nums.length
                if (nums(a) + nums(b) == target)
            }   yield Array(a, b)
        
        indicies(0)
    }
}

File: leetcode/1.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        for (i in nums.indices) {
            for (j in nums.indices) {
                if (i < j && nums[i] + nums[j] == target)
                    return intArrayOf(i, j)
            }
        }
        return intArrayOf()
    }
}
